src/wallet/rpcwallet.cpp

```diff
@@ -5225,6 +5213,25 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
     ThrowIfInitialBlockDownload();
+    const auto chainparams = Params();
+    const auto consensus = chainparams.GetConsensus();
+    int nextBlockHeight = chainActive.Height() + 1;
+
+    // This API cannot be used to create coinbase shielding transactions before Sapling
+    // activation.
+    if (!consensus.NetworkUpgradeActive(nextBlockHeight, Consensus::UPGRADE_SAPLING)) {
+        throw JSONRPCError(
+            RPC_INVALID_PARAMETER, "Cannot create shielded transactions before Sapling has activated");
+    }
+
+    auto strategy =
+        ResolveTransactionStrategy(
+                ReifyPrivacyPolicy(
+                        PrivacyPolicy::AllowRevealedSenders,
+                        params.size() > 4 ? std::optional(params[4].get_str()) : std::nullopt),
+                // This has identical behavior to `AllowRevealedSenders` for this operation, but
+                // technically, this is what “LegacyCompat” means, so just for consistency.
+                PrivacyPolicy::FullPrivacy);
```

Comment:
That’s a braino – it should have been `AllowFullyTransparent`, which is the only established meaning for LegacyCompat that makes sense here.

However, in the context of `z_shieldcoinbase`, `AllowLinkingAccountAddresses` might be the more sensible `LegacyCompat` meaning. I was thinking we could do a similar check as `z_sendmany` – if we’re shielding to a UA, `LegacyCompat` is `AllowRevealedSenders` (or `AllowFullyTransparent`, which has the same meaning here), and if there’s no UA, then `LegacyCompat` is `ALAA`.

But … it’s not quite the same, as (until #6467 is merged) `ALAA` only has meaning when you’re sending from a UA (or worse, `ANY_TADDR`), so conditionalizing on _any_ UA involved is pretty meaningless (we could just always default to `AllowRevealedSenders` and get the same semantics). Also, I’m assuming `LegacyCompat` for `z_sendmany` was added when UAs were added, so introducing it/privacyPolicies wasn’t a breaking change (it maintained the existing semantics for the tx that were possible before UAs), but here it would be a breaking change, as it changes the semantics for UA-involved tx that have already been established (in that previously using `ANY_TADDR` or a UA as the sender would imply `AllowLinkingAccountAddresses`).

Reply:
For now, I’ve made the simple change to use `AllowFullyTransparent`, but I‘d like some help figuring out the correct thing to do.

Reply:
In practice I don't think that there's much difference between `AllowFullyTransparent` and `AllowLinkingAccountAddresses` under `LegacyCompat` so this is fine for now.